src/main/java/seedu/homerce/ui/schedulepanel/SchedulePanel.java

```diff
@@ -0,0 +1,153 @@
+package seedu.homerce.ui.schedulepanel;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.layout.ColumnConstraints;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Region;
+import seedu.homerce.commons.core.LogsCenter;
+import seedu.homerce.model.appointment.Appointment;
+import seedu.homerce.model.service.Duration;
+import seedu.homerce.ui.UiPart;
+
+/**
+ * Panel containing list of Appointments.
+ */
+public class SchedulePanel extends UiPart<Region> {
+    public static final String TAB_NAME = "Schedule";
+    private static final int SINGLE_COLUMN_WIDTH = 120;
+    private static final int ROW_SPAN = 1;
+    private static final int NUM_OF_HALF_HOURS = 48;
+    private static final double SCALE_FACTOR = 2.0;
+    private static final int GRID_INDEX_BUFFER = 1;
+    private static final String FXML = "schedulepanel/SchedulePanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(seedu.homerce.ui.schedulepanel.SchedulePanel.class);
+
+    private final ObservableList<Appointment> appointments;
+
+    @FXML
+    private ScrollPane scrollPane;
+
+    @FXML
+    private GridPane gridPane;
+
+    /**
+     * Creates a {@code AppointmentListPanel} with the given {@code ObservableList}.
+     */
+    public SchedulePanel(ObservableList<Appointment> appointments) {
+        super(FXML);
+        this.appointments = appointments;
+    }
+
+    /**
+     * Fills the grid pane with the slots.
+     */
+    public void construct() {
+        if (appointments.size() == 0) {
+            System.out.println("No appointments in schedule");
```

Comment:
Moving forward, I think it'll be good to have an empty state screen for the user